core/io.rs

```diff
@@ -64,6 +64,53 @@ impl BufView {
     self.cursor = 0;
     old
   }
+
+  /// Adjust the length of the remaining buffer. If the requested size is greater than the current
+  /// length, no changes are made.
+  pub fn truncate(&mut self, size: usize) {
+    match &mut self.inner {
+      BufViewInner::Empty => {}
+      BufViewInner::Bytes(bytes) => bytes.truncate(size + self.cursor),
+      BufViewInner::JsBuffer(buffer) => buffer.truncate(size + self.cursor),
+    }
+  }
+
+  /// Split the underlying buffer. The other piece will maintain the current cursor position while this buffer
+  /// will have a cursor of zero.
+  pub fn split_off(&mut self, at: usize) -> Self {
+    let at = at + self.cursor;
+    assert!(at <= self.len());
+    let other = match &mut self.inner {
+      BufViewInner::Empty => BufViewInner::Empty,
+      BufViewInner::Bytes(bytes) => BufViewInner::Bytes(bytes.split_off(at)),
+      BufViewInner::JsBuffer(buffer) => {
+        BufViewInner::JsBuffer(buffer.split_off(at))
+      }
+    };
+    Self {
+      inner: other,
+      cursor: 0,
+    }
+  }
+
+  /// Split the underlying buffer. The other piece will have a cursor of zero while this buffer
+  /// will maintain the current cursor position.
+  pub fn split_to(&mut self, at: usize) -> Self {
+    assert!(at <= self.len());
+    let at = at + self.cursor;
```

Comment:
Shouldn't we assert that this is also <= self.len?

Reply:
In both the bytes and v8 slice cases, there's a matching assert in the underlying V8Slice/Bytes, so we'll catch those there.